### PR TITLE
83846 disability benefits migrate 0781 uploads to lighthouse

### PR DESCRIPTION
**Pull request by ajones446**
This PR will allow us to progressively transition uploading 0781 forms to the new Lighthouse Documents Benefits API.

All new behavior is hidden behind a killswitch flipper, `disability_compensation_use_api_provider_for_0781_uploads`, which will only use the API provider if turned on.

If the killswitch is enabled, the API Provider Factory itself will decide if we use the Lighthouse or EVSS provider client for the uploads, based on the status of the `disability_compensation_upload_0781_to_lighthouse` flipper.

These changes are very similar to the ones made to the [BDD Instruction upload job](https://github.com/department-of-veterans-affairs/vets-api/pull/18809), which was a much lower risk to do first and is already in production. Similar work is in progress for Veteran uploaded documents as well: https://github.com/department-of-veterans-affairs/vets-api/pull/19182

Note: The submit0781 job processes both forms 0781 and 0781a, if present. A separate PR will be created to test the 0781a option.

## Summary

- *This work is behind a feature toggle (flipper): YES (see above)
- *(Which team do you work for, does your team own the maintenance of this component?)* Disability Benefits team 2, aka Carbs; yes we do own the maintenance of these components
- *(If introducing a flipper, what is the success criteria being targeted?)* A certain amount of submissions will fail on upload to Lighthouse just as they do to EVSS now - it's the same underlying system; however we will be on the look out for new errors that expose bugs on our side or in Lighthouse's benefits documents API.

## Related issue(s)

- *Link to ticket created in va.gov-team repo OR screenshot of Jira ticket if your team uses Jira*
(https://github.com/department-of-veterans-affairs/va.gov-team/issues/83846)

## Testing done

- [x] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*
This feature will be tested extensively on staging before release. It will be rolled out incrementally based on a percentage of production users, starting at 1%. All 0781 upload attempts will be logged and monitored.

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution 
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable) *Not yet, but we have a ticket to do this
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: spec/sidekiq/evss/disability_compensation_form/submit_form0781_spec.rb [hunk 1]
```diff
@@ -170,4 +170,113 @@
...
+            'data' => {
+              'success' => true,
+              'requestId' => lighthouse_request_id
+            }
+          }
+        )
+      end
+
+      context 'when a submission has a 0781 form' do
+        it 'successfully uploads the 0781 document to Lighthouse' do
```
Comment (ajones446) [inline spec/sidekiq/evss/disability_compensation_form/submit_form0781_spec.rb:232]: I will do a follow-up PR to test the 0781 upload in a new context block.

File: lib/disability_compensation/providers/document_upload/lighthouse_supplemental_document_upload_provider.rb [hunk 2]
```diff
@@ -11,7 +11,9 @@ class LighthouseSupplementalDocumentUploadProvider
   # Maps VA's internal Document Types to the correct document_type attribute for a Lighthouse526DocumentUpload polling
   # record. We need this to create a valid polling record
   POLLING_DOCUMENT_TYPES = {
-    'L023' => Lighthouse526DocumentUpload::BDD_INSTRUCTIONS_DOCUMENT_TYPE
+    'L023' => Lighthouse526DocumentUpload::BDD_INSTRUCTIONS_DOCUMENT_TYPE,
+    'L228' => Lighthouse526DocumentUpload::FORM_0781_DOCUMENT_TYPE,
+    'L229' => Lighthouse526DocumentUpload::FORM_0781A_DOCUMENT_TYPE
```
Comment (NB28VT) [inline lib/disability_compensation/providers/document_upload/lighthouse_supplemental_document_upload_provider.rb:16]: Oops I actually ended up changing how this works, I had to do some trickery for the vet uploads, and I ended up removing this list in the pending veteran upload pr that is still in review.

If you could make the change to use [this method](https://github.com/department-of-veterans-affairs/vets-api/pull/19117/files#diff-a0a3ecd0f4e7798e8a50d8a0c7807733ceec98c04dec72c0fd9a4f64d333ef6dR182) instead that should take care of it.

File: lib/disability_compensation/providers/document_upload/lighthouse_supplemental_document_upload_provider.rb [hunk 3]
```diff
@@ -179,9 +181,6 @@ def polling_record_document_type
     # Veteran-uploaded documents can be numerous VA document types
     return Lighthouse526DocumentUpload::VETERAN_UPLOAD_DOCUMENT_TYPE if @supporting_evidence_attachment
 
-    case @va_document_type
-    when 'L023'
-      Lighthouse526DocumentUpload::BDD_INSTRUCTIONS_DOCUMENT_TYPE
-    end
+    POLLING_DOCUMENT_TYPES[@va_document_type]
```
Comment (NB28VT) [inline lib/disability_compensation/providers/document_upload/lighthouse_supplemental_document_upload_provider.rb:184]: Oh nevermind, this is better yeah let's keep this! Sam was right I didn't need the case statement. This is documented better

File: lib/disability_compensation/providers/document_upload/lighthouse_supplemental_document_upload_provider.rb (hunk 2 above)
Comment (NB28VT) [reply lib/disability_compensation/providers/document_upload/lighthouse_supplemental_document_upload_provider.rb:16]: Disregard see below

**Review by NB28VT — DISMISSED** [review]
This looks good to me!

File: app/sidekiq/evss/disability_compensation_form/submit_form0781.rb
Comment (rmtolmach) [inline app/sidekiq/evss/disability_compensation_form/submit_form0781.rb:1]: You probably already know about it, but I was recently shown this [Sidekiq Job Standards](https://github.com/department-of-veterans-affairs/va.gov-team-sensitive/blob/master/teams/benefits/architectural-decision-records/sidekiq-guidance.md) doc. I'm starting to pass it around to make sure people writing sidekiq jobs are aware! 🙂

**Review by rmtolmach — APPROVED** [review]
👍

**Comment by rmtolmach** [comment]
oh... @ajones446 you have some merge conflicts. @ me once they're resolved and I'll re-review.

**Review by NB28VT — APPROVED** [review]
Yeah this looks to me like it survived the merge conflicts just fine!

**Comment by ajones446** [comment]
> oh... @ajones446 you have some merge conflicts. @ me and remove the `merge-conflict` label once they're resolved and I'll re-review.

Thank you @rmtolmach! I resolved the conflicts and removed the label, but now there are some unrelated test failures.

**Pull request merged**